test/functional/rpc_deprecated.py

```diff
@@ -4,12 +4,17 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test deprecation of RPC calls."""
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_raises_rpc_error
 
 class DeprecatedRpcTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 2
         self.setup_clean_chain = True
-        self.extra_args = [[], ["-deprecatedrpc=validateaddress"]]
+        self.extra_args = [[], ["-deprecatedrpc=generate"]]
+
+    def skip_test_if_missing_module(self):
+        # The generate RPC method requires the wallet to be compiled
+        self.skip_if_no_wallet()
```

Comment:
I don't think the whole test should be skipped if there is not wallet compiled in, but rather only the wallet-specific `assert_raises_rpc_error` should be skipped. This way other tests in this file that are not wallet-specific would run when the wallet is not compiled.